leaf-ui/rappid-extensions/BackboneModelsIntention.js

```diff
@@ -7,61 +8,404 @@
+    }, 
+ 
+    /**
+     * Clears all FuncSegments for the Intention's
+     * EvolvingFunction and adds new FuncSegments according to the current
+     * function type.
+     * Called when user changes the function type in Element Inspector 
+     * @param {String} funcType
+     */
+    setEvolvingFunction: function(funcType) {
+
+        this.set('evolvingFunction', new EvolvingFunctionBBM({type: funcType, functionSegList: []}));
+
+        // Since function changed, remove all current absolute constraints related to this intention
+        this.removeAbsConstraint();
+ 
+        // Add new absolute constraints if required
+        this.addAbsConstraint(funcType);
+ 
+        // var initValue = graph.get('userEvaluationList').get(this.cid, '0').get('assignedEvidencePair');
+        var initValue = graph.getUserEvaluationBBM(this.cid, '0').get('assignedEvidencePair');
+ 
+
+        // Creates the correct FunctionSegmentBBM(s) for the selected function type
+        if (funcType == 'C' || funcType == 'R' || funcType == 'I' || funcType == 'D' || funcType == 'UD') {
+            if (funcType == 'C') {
+                var seg =  new FunctionSegmentBBM({type: funcType, refEvidencePair: initValue, startTP: '0', startAT: 0});  
+            } else if (funcType == 'R') {
+                // The reference evidence pair for a Stochastic function is always 0000
+                var seg =  new FunctionSegmentBBM({type: funcType, refEvidencePair: '(no value)', startTP: '0', startAT: 0}); 
+            } else if (funcType == 'I' || funcType == 'D') {
+                var seg =  new FunctionSegmentBBM({type: funcType, refEvidencePair: null, startTP: '0', startAT: 0}); 
+            } else if (funcType == 'UD') {
+                var seg =  new FunctionSegmentBBM({type: 'C', refEvidencePair: initValue, startTP: '0', startAT: 0}); 
+            }
+            this.getFuncSegments().push(seg);
+        } else if (funcType == 'RC' || funcType == 'CR' || funcType == 'MP' || funcType == 'MN' || funcType == 'SD' || funcType == 'DS') {
+            if (funcType == 'RC') {
+                // Stochastic and Constant
+                var seg1 =  new FunctionSegmentBBM({type: 'R', refEvidencePair: '(no value)', startTP: '0', startAT: 0}); 
+                var seg2 =  new FunctionSegmentBBM({type: 'C', refEvidencePair: null, startTP: 'A', startAT: myNull}); 
+            } else if (funcType == 'CR') {
+                // Constant and Stochastic
+                var seg1 =  new FunctionSegmentBBM({type: 'C', refEvidencePair: initValue, startTP: '0', startAT: 0}); 
+                var seg2 =  new FunctionSegmentBBM({type: 'R', refEvidencePair: '(no value)', startTP: 'A', startAT: myNull}); 
+            } else if (funcType == 'MP') {
+                // Increase and Constant
+                var seg1 =  new FunctionSegmentBBM({type: 'I', refEvidencePair: null, startTP: '0', startAT: 0}); 
```

Comment:
Same question about why the first segment has a null refEvidencePair - isn't always the initial value for TP 0? Could be misunderstanding tho